### DISCUSSION TITLE
How to get location to uploaded file to s3

### DISCUSSION THREAD

**Discussion by desoliture** [aws/aws-sdk-java-v2#4577] [Q&A]
Hi
Im creating simple image service using java and spring boot
I need to upload image and return a url to this image
Before we were using Multipart upload with just 1 part, now Im trying to do it with just Transfer Manager and UploadRequest
So, in the CompleteMultipartUploadResponse there's a location() method witch return exactly what I need
Im having troubles finding the same thing for the PutObjectResponse

Code snippet:
```
var requestBody = AsyncRequestBody.forBlockingInputStream(multipartFile.getSize());
var uploadRequest = UploadRequest.builder()
         .requestBody(requestBody)
                .putObjectRequest(builder -> builder.bucket(bucketName).key(key).contentType(multipartFile.getContentType()))
                .addTransferListener(LoggingTransferListener.create())
                .build();
var upload = transferManager.upload(uploadRequest);
requestBody.writeInputStream(inputStream);
var completedUpload = upload.completionFuture().join();
var response = completedUpload.response();
return **response.location()**;
```

**Comment by debora-ito** [comment]
Indeed, `location` is not present in PutObjectResponse. But you can manually construct the S3 URI by using the formula:
```
s3://<bucketName>/<key>
```

**Reply by desoliture** [reply]
Just found this:
`String objectUrl = s3Client.utilities().getUrl(GetUrlRequest.builder()`

Is this a good solution?
